packages/hash/shared/src/types.ts

```diff
@@ -0,0 +1,339 @@
+import { VersionedUri } from "@blockprotocol/type-system-web";
+import slugify from "slugify";
+
+export const frontendDomain = process.env.FRONTEND_DOMAIN ?? "localhost:3000";
+
+/** @todo: figure out how to properly determine the base URL of workspace types */
```

Comment:
If we're going to leave this TODO in should we make an asana task?

Reply:
Probably, but I'm not sure what such a task would be called. We have so much we still need to figure out around workspace URLs, how they are determined, what to do if they change (because the domain has changed), whether they can or cannot be used as the ids of the types, etc. All I'm trying to indicate here is that defaulting to whatever `process.env.FRONTEND_DOMAIN` says is not the solution, and I'm not sure how helpful a task for that is.

Reply:
I don't actually understand why that's not the solution? Isn't that literally what we use for the frontend elsewhere? If that's what the frontend is hosted at, that _**needs**_ to be the base URL of workspace types, surely?

Reply:
One reason might be to make it easier for non-technical users to host HASH we may want to make the domain configurable in a UI, similar to how it works in WordPress (something I think I recall @nonparibus mentioning a while ago).

Reply:
Sure but that's a wider question about figuring out the domain, it's not about this specific part of the code. This code _should_ go through the exact same path as any other piece of code that requires knowledge about the domain or else we risk inconsistencies and a completely broken app

Reply:
Do you want this to become the source of truth for now then? And remove the vercel specific staging environment frontend URL we currently have in the frontend?

Or do you want me to add a todo to unify them to avoid touching this as part of this PR, and potentially removing vercel functionality that may be useful down the line.

Reply:
Fine to leave this as a `@todo` in-code for now.

Reply:
I've rewritten the todo and attached an asana task for fixing the staging environments in https://github.com/hashintel/hash/pull/1193/commits/3bbf30d2608ebdac94ed99deb05c766a43e017bb